Closes softlayer/sl-ember-components#209

juwara0 opened the pull request:
Wrote tests for the sl-datepicker component, updated documentation and removed items mentioned in comments.

notmessenger commented on addon/components/sl-date-picker.js:
```diff
@@ -68,6 +68,13 @@ export default Ember.Component.extend( TooltipEnabled, {
     daysOfWeekDisabled: [],
 
     /**
+     * When true, the datepicker will not be displayed
```
well....the input field will display but be disabled and not able to be interacted with in order to select the date selection calendar.  Perhaps tweak this working just a small bit?

juwara0 replied:
This wording has been updated.

notmessenger commented on addon/components/sl-date-picker.js:
```diff
@@ -166,6 +173,14 @@ export default Ember.Component.extend( TooltipEnabled, {
 
     /**
+     * The placeholder text that the datepicker should show
+     * be disabled
```
`be disabled`?

juwara0 replied:
I have removed this. It was copy-pasted.

notmessenger commented on addon/components/sl-date-picker.js:
```diff
@@ -200,6 +215,13 @@ export default Ember.Component.extend( TooltipEnabled, {
     /**
+     * The date either selected by the datepicker or entered by the user
+     *
+     * @type {?Date}
```
Can this property also be a `String`?

juwara0 replied:
I changed this as it actually should be a string and not a Date.

notmessenger commented:
What is the behavior of the underlying data picker component when either `setEndDate` or `setStartDate` are set via calls to `setEndDate()` and `setStartDate()`?  Does it "redraw" itself, removing a selection that may now be out of the range?  Need to ensure that our expectation that as observers these properties can change that the component utilizing can actually react to such changes (resetting of values) rather than they only being intended to be set once.

notmessenger commented on addon/templates/components/sl-date-picker.hbs:
```diff
@@ -1,11 +1,10 @@
 {{/if}}
 
 {{input
     viewName="sl-date-picker-view"
```
If getting rid of this property in the `label` element than can get rid of its assignment here.

juwara0 replied:
This assignment has been removed.

notmessenger commented on tests/dummy/app/templates/demos/sl-date-picker.hbs:
```diff
@@ -52,6 +48,10 @@
     {{/property-text}}
 
+    {{#property-text name="disabled" type="string" default="false"}}
+        When true, the datepicker will not be displayed.
```
Same comment about this property in the component - make sure to synchronize whatever the new property description becomes.

juwara0 replied:
This wording has been updated.

notmessenger commented on tests/dummy/app/templates/demos/sl-date-picker.hbs:
```diff
@@ -111,6 +115,10 @@
         When true, highlight the current date.
     {{/property-text}}
 
+    {{#property-text name="value" type="date"}}
```
Same inquiry as to the property in the component - can this also be a `String`?

juwara0 replied:
I changed this as it actually should be a string and not a Date.

notmessenger commented on tests/unit/components/sl-date-picker-test.js:
```diff
@@ -0,0 +1,264 @@
+    unit: true
+});
+
+test('it renders', function( assert ) {
```
https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

juwara0 replied:
spacing issue fixed

notmessenger commented on tests/unit/components/sl-date-picker-test.js:
```diff
@@ -0,0 +1,264 @@
+    assert.equal( component._state, 'inDOM' );
+});
+
+test( 'Default class names are present', function( assert ) {
```
Either in this test or different one the classes applied to the ``{{input}}`` in the template need to be ensured as well.

juwara0 replied:
Tests for the classes applied to {{input}} have been added.

notmessenger commented:
The `label` property is not documented anywhere nor is it tested.

notmessenger commented:
The `helpText` property is not documented anywhere nor is it tested.

notmessenger commented:
Let's go ahead and test that the object returned by https://github.com/juwara0/sl-ember-components/blob/209/addon/components/sl-date-picker.js#L321 contains expected keys.

juwara0 commented:
Tests have been written, documentation has been updated and default properties have been assigned for  'label' and 'helpText'.

juwara0 commented:
A test has been added that verifies that the options object contains the expected keys.

juwara0 commented:
Refactored setStartDate() and setEndDate() into updateDateRange() and added tests to ensure that our expectation that as observers these properties can change and that the component utilizing can actually react to such changes (resetting of values) rather than they only being intended to be set once.

notmessenger commented:
https://github.com/juwara0/sl-ember-components/blob/209/addon/components/sl-date-picker.js#L275 needs to be `const`

juwara0 commented:
https://github.com/juwara0/sl-ember-components/blob/209/addon/components/sl-date-picker.js#L275 has been changed to be const
